pipelines/pipeline_katello_34_to_35_upgrade.yml

```diff
@@ -87,6 +116,11 @@
     foreman_installer_scenario: katello
     foreman_installer_options_internal_use_only:
       - "--disable-system-checks"
+      - "--reset-puppet-server-version"
```

Comment:
Any reason you can't use `--upgrade-puppet` here (and below)?

Reply:
I was following [this wiki](http://projects.theforeman.org/projects/foreman/wiki/Upgrading_from_Puppet_4_to_5#Prepare-And-Run-The-Installer), which doesn't mention `--upgrade-puppet` at all. Would you mind updating the wiki to the "@ekohl recommended way"? ;) In the meantime, I'll go ahead and test it out.

Reply:
@ekohl I tried it, and apparently it's unsupported during an upgrade:

```
Concurrent use of --upgrade and --upgrade-puppet is not supported. Please run --upgrade first, then --upgrade-puppet.
```

So I could leave it as is and have only one installer run for the upgrade (and puppet upgrade) or have two installer runs just to use the 'more convenient' option. That doesn't sound more convenient. Maybe I'm missing something?

Reply:
And, here's what I get when trying to use the installer with `--upgrade-puppet` before re-running it with `--upgrade`:

```
TASK [foreman_installer : Run installer] *****************************************************************************************************************************************************************************************************
Thursday 10 May 2018  12:01:49 -0400 (0:00:00.061)       0:01:28.680 **********
fatal: [pipeline-katello-3.4-3.5-centos7]: FAILED! => {
    "changed": true,
    "cmd": "foreman-installer -v --scenario \"katello\"  --disable-system-checks --upgrade-puppet",
    "delta": "0:00:01.087437",
    "end": "2018-05-10 16:01:51.006276",
    "rc": 1,
    "start": "2018-05-10 16:01:49.918839"
}

STDOUT:

Puppet already installed and upgraded. Skipping.


MSG:

non-zero return code

fatal: [pipeline-proxy-3.4-3.5-centos7]: FAILED! => {
    "changed": true,
    "cmd": "foreman-installer -v --scenario \"katello\"  --disable-system-checks --upgrade-puppet",
    "delta": "0:00:02.131133",
    "end": "2018-05-10 16:01:52.079101",
    "rc": 27,
    "start": "2018-05-10 16:01:49.947968"
}

STDOUT:

[ INFO 2018-05-10 16:01:52 verbose] Executing hooks in group pre_migrations
[ INFO 2018-05-10 16:01:52 verbose] All hooks in group pre_migrations finished
[ERROR 2018-05-10 16:01:52 verbose] You are trying to replace existing installation with different scenario. This may lead to unpredictable states. Use --force to override. You can use --compare-scenarios to see the differences
[ INFO 2018-05-10 16:01:52 verbose] Log was be written to /var/log/foreman-installer/katello.log
[ INFO 2018-05-10 16:01:52 verbose] Logs flushed


MSG:

non-zero return code
```

Not sure why it thinks this is a different scenario...